src/app/descriptors/budget/budget4dig/index.ts

```diff
@@ -1,11 +1,14 @@
 import { SimpleDescriptor } from '../../../model';
 
 export default new SimpleDescriptor({
-  pathPrefix: 'budget/00[0-9]{4}/',
+  pathPrefix: 'budget/00([^0][0-9]{3}|[0-9][^0][0-9]{2})/',
```

Comment:
instead of making these regex monstrosities, simply add the income descriptors before the budget descriptors so the y match first.